src/vaults/BalanceSheet.sol

```diff
@@ -97,23 +99,36 @@ contract BalanceSheet is Auth, Recoverable, IBalanceSheet, IBalanceSheetGatewayH
         external
         authOrManager(poolId)
     {
-        AssetId assetId = poolManager.assetToId(asset, tokenId);
-        _noteDeposit(poolId, scId, assetId, asset, tokenId, msg.sender, amount);
-        _executeDeposit(poolId, asset, tokenId, msg.sender, amount);
+        noteDeposit(poolId, scId, asset, tokenId, msg.sender, amount);
```

Comment:
With the simplifications in the rest, I would say the internal methods are rather misdirection, and it is simpler to read to inline everything.